### ISSUE TITLE
SLDSButton: button missing button-name

### ISSUE THREAD

**Issue by jcjuett** [salesforce/design-system-react#1894]
Bug at this link: https://design-system-react-co-pr-1767.herokuapp.com/?selectedKind=SLDSButton&selectedStory=Small%20Icon%20Hint%20inverse&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

Reproduce error: tab to button and detect button label ("unlabeled 0 button") using VoiceOver.
HTML: `<button class="slds-button slds-button_icon-border-inverse slds-button_icon-small" type="button"><svg aria-hidden="true" class="slds-button__icon slds-button__icon_inverse-hint"><use xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#down"></use></svg></button>`

Success criteria: Button name ("Small Icon Hint inverse") should be read. Change HTML above to: <button class="slds-button slds-button_icon-border-inverse slds-button_icon-small" type="button"><svg aria-hidden="true" class="slds-button__icon"><use xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#down"></use></svg><span class="slds-assistive-text">Small Icon Hint inverse</span></button>

**Comment by pulkonet** [comment]
I will take up this issue.

**Comment by pulkonet** [comment]
There is a very slight error in the code, thanks for pointing it out @jcjuett I have fixed it and I am submitting a PR.
